src/structure/options.ts

```diff
@@ -67,7 +66,11 @@ export class StructureOptions extends OptionsGroup {
     // Callback to get the initial positioning of the settings modal.
     private _positionSettingsModal: PositioningCallback;
 
-    constructor(root: HTMLElement, positionSettings: PositioningCallback, properties?: MapData) {
+    constructor(
+        root: HTMLElement,
+        positionSettings: PositioningCallback,
+        propertiesName?: string[]
```

Comment:
```suggestion
        propertiesName: string[] = []
```

you can setup different default values than undefined